pkgs/development/python-modules/libmambapy/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+, cmake
+, ninja
+, libmamba
+, python
+, pybind11
+, setuptools
+, wheel
+, scikit-build
+, breakpointHook
+, ...
+}:
+buildPythonPackage rec {
+  pname = "libmambapy";
+  version = "1.5.7";
+  pyproject = true;
+  nativeBuildInputs = [ cmake ninja breakpointHook ];
+  buildInputs = [ python libmamba pybind11 ];
```

Comment:
Also native?

Reply:
I thought `nativeBuildInputs` was only used for runtime executables, like cmake, ninja?
Those specific packages are only needed at build time to link against. 

In this case, python is not executed. The python libaries/headers are only used for linking with the c++ binary.

Reply:
> I thought nativeBuildInputs was only used for runtime executables

Yes, it's for the code that is to be executed during the build/on the build platform.

Python is executed by pybind11's cmake rules, and I think so is pybind11... As to whether `pkgsBuildHost.python3Packages.python` would come with the host's libpython... I'm really not sure. Idk how much of python-for-cross is actually functional

People have been putting `pybind11` in `nativeBuildInputs` afaik (e.g. looking now at `torch/default.nix`)

Reply:
Actually, I think both pythonForBuild and pythonForHost should be passed on into the build environment by `buildPythonPackage`. Did you try removing `python` from the list?

Reply:
-  pybind11 is a header only and it doesn't require the use of python to generate intermediate types. In cmake, python is only needed by by pybind11 to properly set the output library name. I've never cross compiled, so there's maybe something that I don't understand.

- Removing python worked.

- When I moved `libmamba` to `nativeBuildInputs`, I got this error:

```
CMake Error at libmambapy/CMakeLists.txt:15 (find_package):
  By not providing "Findlibmamba.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "libmamba",
  but CMake did not find one.

  Could not find a package configuration file provided by "libmamba" with any
  of the following names:

    libmambaConfig.cmake
    libmamba-config.cmake
```
I don't understand why this happens.

Reply:
Libmamba was placed in `buildInputs` correctly, I'm pretty sure you just link the library to be executed on the host platform (i.e. after the build). As for the new error, I suppose it's because of `targetOffset` in https://github.com/NixOS/nixpkgs/blob/f471be9644f3ab2f3cb868de1787ab70a537b0e7/pkgs/by-name/cm/cmake/setup-hook.sh#L146, but I'm not sure and it's not relevant either

Reply:
@SomeoneSerge  I'm not sure what to make of your last comment. Is it fine as is or do I need to move it `nativeBuildInputs` and troubleshoot the error?

Reply:
No-no, libmamba goes in buidlInputs, you did it right